File: Python/BinarySearchTreeDFS.py
class Node:
    data = 0
    left = None
    right = None

def search(current: Node, value: int) -> bool:
    if current == None:
        return False
    if current == value:
        return True
    if current.data < value:
        return search(current.right, value)
    return search(current.left, value)

def dfs(head: Node, value: int) -> bool:
    return search(head, value)

File: Java/BinarySearchTreeDFS.java
public class BinarySearchTreeDFS {
    Node head;
    public class Node{
        int data;
        Node left;
        Node right;

        public Node(int value){
            data = value;
            left = null;
            right = null;
        }
    }

    public boolean search(Node current, int value){
        if(current == null){
            return false;
        }

        if(current.data == value){
            return true;
        }

        if(current.data < value){
            return search(current.right, value);
        }

        return search(current.left, value);
    }

    public boolean depthFirstSearch(int value){
        return search(head, value);
    }
}
